content/blog/understanding-the-dom/index.md

```diff
@@ -0,0 +1,519 @@
+
+For example, we can [use `data` attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) in order to read and write values via attributes to any given element.
+
+```html
+<!-- index.html -->
+<ul id="list" data-listitems="2">
+  <li>List item 1</li>
+  <li>List item 2</li>
+</ul>
+```
+
+```javascript
+// index.js
+const listEl = document.querySelector('#list');
+console.log(listEl.dataset.listitems); // '2'
+listEl.dataset.listitems = 3;
+console.log(listEl.dataset.listitems); // '3'
+```
+
+![Demonstrating that dataset values are able to be read and written](list_dataset.png)
+
+If you'll notice, I wrote the string `'3'` instead of the numerical value `3` in the code sample's outputs in the comments despite using the numerical `3` to set the value. This behavior is due to how default non-string values are saved to attributes.
+
+By default, the primitive's `toString` will be called to store values.
+
+```javascript
+element.dataset.userInfo = {name: "Tony"}
+console.log(element.dataset.userInfo) // "[object Object]"
+/**
+ * "[object Object]" is because it's running `Object.prototype.toString()`
+ * to convert the object to a string to store on the attribute
+ */
+```
+
+> If you're having a difficult time understanding why `toString` is bring ran or what `prototype` is doing here, don't worry: You're in good company. The JavaScript prototype system is complex and can be difficult to follow.
+>
+> For now, it will suffice to just know that you're only able to store strings in element attribute.
+
+
+
+## Events {#events}
+
+Just as your browser uses the DOM to know where to place elements to display for the user and metadata about the elements in order to interact with screenreaders as-expected, your browser also utilizes the DOM for knowing how to handle user interactions. The way your browser handles user interaction is by listening for _events_ that occur when the user takes action or other noteworthy changes occur.
+
+For example, when you have a form that includes a default button, when that button is pressed it will fire a `submit` event that will then _bubble_ up the DOM tree until it finds a [`<form>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form). This element will by default then run a [`GET` HTML request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET) to the server once it recieves a `submit` event.
+
+![The bubble flow of the `submit` event](./submit_form.svg)
+
+_Bubbling_, as shown here, is the default behavior of any given event. It reflects it's behavior of a given event moving up the DOM tree to the nodes above it as parents. Parent nodes can respond to these events as expected, stop their upward motion on the tree, and more.
```

Comment:
```suggestion
_Bubbling_, as shown here, is the default behavior of all events. It reflects the behavior of the given event, moving up the DOM tree to the parent nodes above the element the event originated from. Parent nodes can respond to these events as expected, stop their upward motion on the tree, and more.
```
- This is my best guess on how to make this paragraph more readable. The second sentence, in particular, wasn't clear to me.

Reply:
```suggestion
_Bubbling_, as shown here, is the default behavior of any given event. Its behavior is to move an event up the DOM tree to the nodes above it, moving from child to parent until it hits the root. Parent nodes can respond to these events as expected, stop their upward motion on the tree, and more.
```

I'd like to leave the first sentence as-is